File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.py
def solution(cacheSize, cities):
    cache = []
    answer = 0
    if cacheSize == 0:
        return 5 * len(cities)
    for city in cities:
        city = city.lower()
        if city in cache:
            del cache[cache.index(city)]
            cache.append(city)
            answer += 1
        else:
            if len(cache) + 1 > cacheSize:
                cache.pop(0)
            cache.append(city)
            answer += 5
    return answer

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.js
function solution(cacheSize, cities) {
    let cache = [];
    let runtime = 0;
    if (cacheSize == 0) return 5 * cities.length;
    return cities.reduce((acc, city) => {
        city = city.toLowerCase();
        let idx = cache.indexOf(city);
        if (idx > -1) {
            cache.splice(idx, 1);
            cache.push(city);
            return acc + 1;
        }
        if (cache.length + 1 > cacheSize) cache.shift();
        cache.push(city);
        return acc + 5;
    }, 0)
}